### ISSUE TITLE
Unable to connect to a gRPC Server hosted in Kestral as HTTPS from a gRPC C++ Client using default certificates (Windows 10)

### ISSUE THREAD

**Issue by msbasanth** [grpc/grpc#26658]
### What version of gRPC and what language are you using?
gRPC v1.38.1
C++ Client, C#(ASP.Net Core) Server

### What operating system (Linux, Windows,...) and version?
Windows 10

### What runtime / compiler are you using (e.g. python version or version of gcc)
Visual Studio 2019 C++ Compiler

### What did you do?
I am using ASP.Net Core (Grpc.Net) for creating a HTTPS gRPC server hosted in Kestral. The communication between C# Client to the server (HTTPS) works fine with out adding any certificate.
It looks like they are using the default certificates for communication.

Now I have a C++ gRPC Client in **Windows 10** and trying to connect to same server from client, the endpoint is https://localhost:50051. 

This is my Kestral configuration @server.
```
webBuilder.ConfigureKestrel(serverOptions =>
                    {
                        serverOptions.Listen(IPAddress.Any, 50051, listenOptions =>
                       {
                           listenOptions.Protocols = Microsoft.AspNetCore.Server.Kestrel.Core.HttpProtocols.Http2;
                           listenOptions.UseHttps();

                       });
                    }).UseStartup<Startup>();
```
As you could see I am not using certificates in the server (wanted to use default certificates) the same way I used C# gRPC Client.

i.e. Connection to server works using C# Client

```
var channel = GrpcChannel.ForAddress("https://localhost:50051");
ecgDataClient = new Data.DataClient(channel);
```
But when I tried to connect using a C++ gRPC Client it fails.
### What did you expect to see?
Able to connect to Kestral gRPC Server using default certificates.

### What did you see instead?

But with C++ gRPC Client Client I am unable to connect (tried both InSecure & SslCredentials),

```
auto channel_creds = grpc::SslCredentials(grpc::SslCredentialsOptions());
DataGrpcClient grpcClient( grpc::CreateChannel("localhost:50051", channel_creds));
```

With grpc::SslCredentials(grpc::SslCredentialsOptions()) it gives below error,

```
E0709 19:46:20.488000000  6724 ssl_utils.cc:570] load_file: {"created":"@1625840180.488000000","description":"Failed to load file","file":"D:\DEV\vcpkg\buildtrees\grpc\src\17cc203898-db2679e7f2.clean\src\core\lib\iomgr\load_file.cc","file_line":72,"filename":"/usr/share/grpc/roots.pem","referenced_errors":[{"created":"@1625840180.488000000","description":"No such file or directory","errno":2,"file":"D:\DEV\vcpkg\buildtrees\grpc\src\17cc203898-db2679e7f2.clean\src\core\lib\iomgr\load_file.cc","file_line":45,"os_error":"No such file or directory","syscall":"fopen"}]}
E0709 19:46:20.509000000  6724 ssl_security_connector.cc:413] Could not get default pem root certs.
E0709 19:46:20.512000000  6724 secure_channel_create.cc:108] Failed to create secure subchannel for secure name 'localhost:50051'
E0709 19:46:20.517000000  6724 secure_channel_create.cc:50] Failed to create channel args during subchannel creation.
E0709 19:46:20.521000000  6724 ssl_security_connector.cc:413] Could not get default pem root certs.
E0709 19:46:20.525000000  6724 secure_channel_create.cc:108] Failed to create secure subchannel for secure name 'localhost:50051'
E0709 19:46:20.529000000  6724 secure_channel_create.cc:50] Failed to create channel args during subchannel creation.
```

### Anything else we should know about your project / environment?
I am trying localhost in Windows 10.

**Comment by ZhenLian** [comment]
Yeah, in gRPC C++, the default location will be pointed to a place defaults to POSIX systems, meaning it doesn't automatically work on the Windows Systems. To solve that, you will need to manually load the trust bundles from that Windows location, and specify them through SslCredentialsOptions.
The reason why it works for ASP.Net is probably because it does some customizations and just specify the default Windows location on top of the core. It makes sense since I guess most of .Net users will use gRPC on Windows systems? It might be helpful to have some folks working on .Net to confirm that(@markdroth do you know any of them)?
You might also find the following useful:
[1] https://github.com/grpc/grpc/issues/26106
[2] https://github.com/grpc/grpc.io/pull/763

**Comment by erm-g** [comment]
@msbasanth did the [suggestion](https://github.com/grpc/grpc/issues/26658#issuecomment-878670808) solve your problem?

**Comment by BranigansLaw** [comment]
For anyone caught on this, someone added a fix for this with full code:

https://stackoverflow.com/questions/71580681/ssl-grpc-client-works-fine-in-c-but-fails-with-unavailable-empty-update-in-c

**Comment by gtcooke94** [comment]
Doing some cleanup, I believe this a replication of https://github.com/grpc/grpc/issues/25533 and was fixed per that issue.
